project/SparkBuild.scala

```diff
@@ -1027,6 +1027,11 @@ object TestSettings {
       }.getOrElse(Nil): _*),
     // Show full stack trace and duration in test cases.
     testOptions in Test += Tests.Argument("-oDF"),
+    // Show only the failed test cases in github action to make the log more readable.
+    testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest,
+      sys.env.get("GITHUB_ACTIONS").map { _ =>
+        Seq("-eNCXEHLOPQMDSF")
```

Comment:
Check https://www.scalatest.org/user_guide/using_the_runner for the options here.

Reply:
It seems that you explicitly enabled all available options except `W` and `U`. If then, could you describe the reason why you choose some and exclude those two, please?
```
W - without color
U - unformatted mode
```

Reply:
Also, could you add the link into the comment at below line 1030 because this is non-trivial?

Reply:
Here we drop all the following events:
```
N - drop TestStarting events
C - drop TestSucceeded events
X - drop TestIgnored events
E - drop TestPending events
H - drop SuiteStarting events
L - drop SuiteCompleted events
O - drop InfoProvided events
P - drop ScopeOpened events
Q - drop ScopeClosed events
R - drop ScopePending events
M - drop MarkupProvided events
```
and enable the following two mode:
```
D - show all durations
F - show full stack traces
```

Reply:
The following two modes will make the readability worse
```
W - without color
U - unformatted mode
```

Reply:
Could you put that into the PR description, please? PR description will become a permanent commit log . So, it will help other people to understand your decision. Thanks, @gengliangwang .

Reply:
@dongjoon-hyun Sure, I have updated the pr description.